src/components/LendModals/index.tsx

```diff
@@ -0,0 +1,517 @@
+                  {asset.underlyingSymbol}
+                </p>
+              )}
+            </Box>
+            <Box
+              mt={2}
+              className={`lendModalInput ${inputFocused ? 'focused' : ''}`}
+            >
+              <Box>
+                <NumericalInput
+                  placeholder={'0.00'}
+                  value={inputValue}
+                  onFocus={() => setInputFocused(true)}
+                  onBlur={() => setInputFocused(false)}
+                  onUserInput={setInputValue}
+                />
+                <p className='span text-secondary'>
+                  ({midUsdFormatter(asset.usdPrice * Number(value))})
+                </p>
+              </Box>
+              <Box
+                className='lendMaxButton'
+                onClick={() => {
+                  setValue((maxAmount ?? 0).toString());
+                }}
+              >
+                {t('max')}
+              </Box>
+            </Box>
+            <Box my={3} className='lendModalContentWrapper'>
+              {!borrow ? (
+                <>
+                  <Box className='lendModalRow'>
+                    <p>{t('suppliedBalance')}:</p>
+                    <p>
+                      {`${supplyBalance.toLocaleString()} ${
+                        asset.underlyingSymbol
+                      }`}
+                      {showArrow && (
+                        <>
+                          <ArrowForward fontSize='small' />
+                          {`${convertBNToNumber(
+                            updatedAsset.supplyBalance,
+                            asset.underlyingDecimals,
+                          ).toLocaleString()} ${asset.underlyingSymbol}`}
+                        </>
+                      )}
+                    </p>
+                  </Box>
+                  <Box className='lendModalRow'>
+                    <p>{t('supplyapy')}:</p>
+                    <p
+                      className={supplyAPY > 0 ? 'text-success' : 'text-error'}
+                    >
+                      {supplyAPY.toLocaleString()}%
+                      {updatedAsset && updatedAPYDiffIsLarge && (
+                        <>
+                          <ArrowForward fontSize='small' />
+                          {`${updatedSupplyAPY.toLocaleString()}%`}
+                        </>
+                      )}
+                    </p>
+                  </Box>
+                  <Box className='lendModalRow'>
+                    <p>{t('borrowLimit')}:</p>
+                    <p>
+                      {midUsdFormatter(borrowLimit)}
+                      {showArrow && (
+                        <>
+                          <ArrowForward fontSize='small' />
+                          {midUsdFormatter(updatedBorrowLimit)}
+                        </>
+                      )}
+                    </p>
+                  </Box>
+                  <Box className='lendModalRow'>

```

Comment:
Theres so much copied code here I don't under stand why we don't just supply an array of the names to and assets

`items = [{name:'totalDebtBalance', value: asset.borrowBalanceUSD]
items.map((item, index) => {
   <Box className='lendModalRow'>`